R/fmi2_interactive.R

```diff
@@ -0,0 +1,340 @@
+#' @title Interactive function for fmi2
+#' @description Interactive function for using the different functions of `fmi2` package.
+#'
+#' @details The function allows user to interactively get observations from different `fmi2`
+#' functions. The function also allows printing dataset citation and the code used for
+#' the function call.
+#'
+#' @importFrom utils menu capture.output
+#'
+#' @return sf object in a long (melted) form.
+#' @export
+#'
+#' @examples
+#'   \dontrun{
+#'     y <- fmi2_interactive()
+#'   }
+fmi2_interactive <- function(){
+
+  # Selecting which observations user wants
+  obs_type <- switch(
+    menu(c("Weather observations", "Temperature observations", "Wind observations",
+           "Precipitation observations", "Airquality observations"),
+         title = "Which observations do you want?") + 1,
+    return(invisible()),
+    "obs",
+    "temp",
+    "wind",
+    "prec",
+    "air"
+  )
+
+  # Selecting spatial resolution
+  if (obs_type %in% c("obs", "temp", "prec")) {
+    hourly <- FALSE
+    daily <- FALSE
+    monthly <- FALSE
+    switch(
+      menu(c("Hourly", "Daily", "Monthly"),
+           title = "Select spatial resolution") + 1,
+      return(invisible()),
+      hourly <- TRUE,
+      daily <- TRUE,
+      monthly <- TRUE
+    )
+  }
+
+  # Selecting starttime and endtime
+  times_ok <- NULL
+  while (is.null(times_ok)) {
+    starttime <- readline(prompt = "Give start time in ISO-format: ")
+    endtime <- readline(prompt = "Give end time: ")
+    # Check time arguments are valid
+    if (!valid_time(starttime, endtime)) {
+      message("Plase give time in right format, yyyy-mm-dd")
+    } else if (valid_time(starttime, endtime)) {
+      times_ok <- TRUE
```

Comment:
It would be good if ISO-format would be explained already on row 50 prompt and not only when the input is wrong (row 54)

Reply:
If the user inputs a correct ISO-format time in start time and invalid time in endtime, the step shouldn't in my opinion return to the beginning (asking for start time), but only ask for the incorrectly inputed end time.

Maybe there could be a tip on how to exit the interactive function if the user becomes frustrated? Put `(press esc to exit interactive function)` everywhere?

Reply:
Also I was thinking that is it really necessary to input the end time? Often this could be left empty if the user wants to have the observations up to the time of usage, i.e. current system time.

Reply:
> Also I was thinking that is it really necessary to input the end time? Often this could be left empty if the user wants to have the observations up to the time of usage, i.e. current system time.

The problem is that the API can only return values for 7 day interval for hourly observations, 372 day interval for daily observations or 3650 day interval for monthly observations. If user only specified start time they would often get error about too long time interval. Maybe the values could be given from the max interval if user only provides start time.

Reply:
I think that's a good solution.